readthedocs/rtd_tests/tests/test_doc_building.py

```diff
@@ -837,6 +839,229 @@ def test_command_oom_kill(self):
             u'Command killed due to excessive memory consumption\n')
 
 
+class TestPythonEnvironment(TestCase):
+
+    def setUp(self):
+        self.project_sphinx = get(Project, documentation_type='sphinx')
+        self.version_sphinx = get(Version, project=self.project_sphinx)
+
+        self.project_mkdocs = get(Project, documentation_type='mkdocs')
+        self.version_mkdocs = get(Version, project=self.project_mkdocs)
+
+        self.build_env_mock = Mock()
+
+    def test_install_core_requirements_sphinx(self):
+        python_env = Virtualenv(
+            version=self.version_sphinx,
+            build_env=self.build_env_mock,
+        )
+        python_env.install_core_requirements()
+        requirements = [
```

Comment:
Use the same list of reqs (common) for both project (created in `setUp`) and append the ones that are specific for documentation type in each test.

This will be easier to follow and to change when we upgrade the reqs also.

Reply:
```
>>> import itertools
>>> d = ['a', 'b']
>>> r = [1, 2]
>>> iter
iter(      itertools  
>>> list(itertools.product(d, r))
[('a', 1), ('a', 2), ('b', 1), ('b', 2)]
>>> for _d in d:
...     for _r in r:
...         print(_d, _r)
... 
a 1
a 2
b 1
b 2
>>> 
````